pkg/cosign/kubernetes/webhook/validator.go

```diff
@@ -138,7 +139,30 @@ func (v *Validator) validatePodSpec(ctx context.Context, ps *corev1.PodSpec, opt
 				continue
 			}
 
-			if err := valid(ctx, ref, keys, ociremote.WithRemoteOptions(remote.WithAuthFromKeychain(kc))); err != nil {
+			tempKeys := keys
+			config := config.FromContext(ctx)
+			if config != nil {
+				authorities, err := config.ImagePolicyConfig.GetAuthorities(ref.Name())
```

Comment:
Question: Wondering how I could structure a unit-test to mock out `ctx` such that `config.FromContext(ctx)` returns a config with mocked data for `authorities[].key.data`. Currently, I tried to use `context.WithValue(ctx, config.CfgKey{}, <ImagePolicyConfig populated struct>)` and this did not work (I made `cfgKey{}` public in `store.go` to temporarily try this).

@dlorenc @vaikas

Reply:
Yup, pulling out the logic into a function and going to test that specifically. I will be passing in a mocked config now so I can isolate the testing to the logic there. 

As for testing the integration of that function, I'll look into that further afterwards.

Reply:
I think using this should work for this:
https://github.com/sigstore/cosign/blob/main/pkg/apis/config/store.go#L53

Reply:
Oh my, how did I not see that... Thanks, will give it a try!

Reply:
It looks good to me.